provider/queue/queue_test.go

```diff
@@ -2,22 +2,28 @@ package queue
 
 import (
 	"context"
+	"fmt"
+	"math/rand"
 	"testing"
 	"time"
 
+	"github.com/ipfs/go-block-format"
 	"github.com/ipfs/go-cid"
 	"github.com/ipfs/go-datastore"
 	"github.com/ipfs/go-datastore/sync"
-	"github.com/ipfs/go-ipfs-blocksutil"
+	"github.com/ipfs/go-ipfs-util"
 )
 
-var blockGenerator = blocksutil.NewBlockGenerator()
-
 func makeCids(n int) []cid.Cid {
 	cids := make([]cid.Cid, 0, n)
 	for i := 0; i < n; i++ {
-		c := blockGenerator.Next().Cid()
-		cids = append(cids, c)
+		data := make([]byte, rand.Intn(256))
+		rand.Read(data)
+		b, err := blocks.NewBlockWithCid(data, cid.NewCidV1(cid.Raw, util.Hash(data)))
+		if err != nil {
+			fmt.Println("This should not have happened.")
+		}
+		cids = append(cids, b.Cid())
```

Comment:
@Stebalien this change was required because the block generater creates blocks with v1 cids and base32 encoding a v1 cid fails (https://github.com/ipfs/go-cid/blob/master/cid.go#L388). This begs the question, is it ok to do this encoding if it will fail for v0 cids? It's unclear to me whether or not this code might run into a v0 cid.

Reply:
Maybe we `Base58BCT` if v0 and `Base32` if v1?

Reply:
Ah. Damn, I forgot about that. Case-sensitive CIDs _should_ be fine (and we can always change this later without any issues as we're just using this for uniqueness).